src/TestUtils/src/DeviceTests/AssertionExtensions.Android.cs

```diff
@@ -155,8 +155,23 @@ public static async Task ShowKeyboardForView(this AView view, int timeout = 1000
 			try
 			{
 				await view.FocusView(timeout);
+				await Task.Yield();
 				KeyboardManager.ShowKeyboard(view);
-				await view.WaitForKeyboardToShow(timeout);
+				await Task.Yield();
+
+				bool result = await Wait(() =>
+				{
+					bool isVisible = KeyboardManager.IsSoftKeyboardVisible(view);
+
+					if (!isVisible)
+						KeyboardManager.ShowKeyboard(view);
```

Comment:
Do we need to reset isVisible here as well?

If isVisible was false before, and then we show the keyboard, then the keyboard becomes visible, so I assume isVisible should be updated before returning it?

Reply:
changed